etl/src/replication/client.rs

```diff
@@ -626,24 +638,43 @@ impl PgReplicationClient {
         publication: Option<&str>,
     ) -> EtlResult<Vec<ColumnSchema>> {
         let (pub_cte, pub_pred) = if let Some(publication) = publication {
-            (
-                format!(
-                    "with pub_attrs as (
-                        select unnest(r.prattrs)
-                        from pg_publication_rel r
-                        left join pg_publication p on r.prpubid = p.oid
-                        where p.pubname = {publication}
-                        and r.prrelid = {table_id}
+            let is_pg14_or_earlier = self.get_server_version().await.unwrap_or(0) < 150000;
```

Comment:
Can we use an `enum` to represent versions? Even if the variants are like: `Pg14OrEarlier`. This way we can make it clearer.

Also, is the `unwrap_or(0)` expected behavior? Do we want to default to `is_pg14_or_earlier`? If we can't grab the Postgres version I would throw an error since it's required for proper functioning of core features.

Reply:
Connected to this, one might argue we could also avoid using lazy initialization via `OnceCell` and maybe just validate that the Postgres version is available on connection creation, that way we fail fast. What do you think?

Reply:
These are not blockers, it's mostly nits.